### PR TITLE
z*: test with assert_path_exists, update heredoc

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by chenrui333 — APPROVED** [review]

File: Formula/z/zigmod.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ def install
   end
 
   test do
-    (testpath/"zig.mod").write <<~EOS
+    (testpath/"zigmod.yml").write <<~YAML
```
Comment (chenrui333) [inline Formula/z/zigmod.rb:46]: why this change, this should be `zig.mod`?

Comment (cho-m) [reply Formula/z/zigmod.rb:46]: Latest versions use `zigmod.yml` - https://github.com/nektro/zigmod/commit/2431edfc4f3987c4c31909713ccf643d2253b02d

Comment (chenrui333) [reply Formula/z/zigmod.rb:46]: I see, thanks!
